repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -270,19 +292,23 @@ def get_repositories_blacklisted():
     return repos_blacklisted
 
 
-def map_repositories(to_install):
-    """Map repositories from PES data to RHSM repository id"""
+def map_repositories(packages):
+    """
+    Map repositories from PES data to RHSM repository id
+
+    :param packages: A dictionary in format {<pkg_name>: <repository>}
+    """
     repo_not_mapped_pkgs = set()
-    for pkg, repo in to_install.items():
+    for pkg, repo in packages.items():
         if repo not in REPOSITORIES_MAPPING:
             repo_not_mapped_pkgs.add(pkg)
-            del to_install[pkg]
+            del packages[pkg]
```

Comment:
here you are also iterating over packages and delete it in the same go, this is not good. Rule of thumb Never delete anything from what you are iterating over.

Reply:
Good catch. I missed that completely.

Reply:
Not exactly I but the code author, but I get you. I'll change it.

Reply:
The same issue was in the `filter_out_pkgs_in_blacklisted_repos` function. I fixed both.